skyportal/handlers/api/news_feed.py

```diff
@@ -0,0 +1,59 @@
+from sqlalchemy import desc
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import DBSession, Source, Photometry, Comment, GroupSource
+
+
+class NewsFeedHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
+        ---
+        description: Retrieve summary of recent activity
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - Success
+                    - type: object
+                      properties:
+                        comments:
+                          type: arrayOfComments
+                          description: Newest comments
+                        sources:
+                          type: arrayOfSources
+                          description: Newest updated sources
+                        photometry:
+                          type: arrayOfPhotometry
+                          description: Newest photometry entries
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        if (self.current_user.preferences and 'newsFeed' in self.current_user.preferences
+            and 'numItemsPerCategory' in self.current_user.preferences['newsFeed']):
+            n_items = int(self.current_user.preferences['newsFeed']['numItemsPerCategory'])
```

Comment:
we need to put a cap on these.  `max(n_items, 20)` or something.

Reply:
Done.